R/prepare_stop_level_data.R

```diff
@@ -26,146 +29,347 @@ granulate_to_stop <- function(mbbs) {
+      "; singing from trees in grassy field at 35.6383,-79.0035" = "",
+      "(stop 3, )?heard at 3rd stop( )?-- it was hard to mistake." = "stop 3",
+      "18 were perched at 14 on power lines" = "",
+      "This is a breeding bird survey route with 20 stops." = "",
+      "including 3 chasing each other!" = "",
+      "this was over 10 miles on traveling count" = "",
+      "over a 10 mile area" = "",
+      "I had 2 additional vireos that I thought sounded like.*" = "",
+      "3 white geese or ducks, walking.*" = "",
+      "3 hens with 12 young birds!.*" = "",
+      "Singing from field on south side of road, heard.*" = "",
+      "Nesting in Eastern chimney of 1407 Baptist road" = "",
+      "2 adults 5 young" = "",
+      "Counted 2 groups of 5 with a couple additional birds" = "",
+      "All 8 were roosting on powerline" = "",
+      "One Horned Lark singing and walking.*" = "",
+      "22 on wire at stop 17 (new school)" = "",
+      "flying and twittering at 20" = "st 20",
+      "heard at stop 17 on Mann's Chapel Road" = "st 17",
+      "One flyover flock.*" = "",
+      "Durham Mini Breeding Bird Survey, 20 stops" = ""
+      )) 
+}
+
+#' Use when species comments follows ,,,,,,1,,,, format
+#' eg. where commas seperate counts at each stop. 
+#' Takes a string and returns a list of 20 numbers
+#' @param x a string from the species_comment column formatted with commas
+#' @param count an integer from the count column of the mbbs
+sp_com_comma_seperated <- function(x, count = -99) {
+  
+  assertthat::assert_that(
+    str_detect(x, ",") # check has commas as format requires
+  )
+  
+  # split based on commas
+  x %>% 
+    stringr::str_split_1(",") %>%
+    str_replace_all(
+      c( # Convert empty strings 0
+        "^$" = "0",
+        # Remove the Stops = / Stops:
+        "Stop(s)?( )?(=)?(:)?( )?" = "",
+        # If it starts with st, drop
+        "( )?st(op)?( )?" = ""
+      )) %>%
+  # make standard list of 20
+  # 2 cases of 19 stops - minor dif in birds on either side of quarter route
+  # leave as is.
+  # Cases with 21st stop where it comes from extra comma after 20th stop
+    pad_or_truncate() %>%
+    as.integer() %>%
+    (\(out) {
+      assertthat::assert_that(
+        length(out) == 20
+        #sum(out) == count #this check is no longer done here.
+        #additional checks go here.
+      )
+      out # return
+    })
+  
+}
+
+#' Use when species comments follows 1=5 or st 1=5 to stop 1=5 format
+#' Takes a string and returns a list of 20 numbers
+#' @param x a string from the species_comment column formatted by "stop=count"
+#' @param count an integer from the count column of the mbbs
+#' @importFrom stringr str_split_1 str_detect str_replace_all
+#' @importFrom assertthat assert_that
+#' @returns out a standardized list of 20 integers, representing the 
+#'    species count at 20 stops.
+sp_com_stop_equals_count <- function(x, count = -99) {
```

Comment:
Why the unused `count` argument? Some question for `sp_com_comma_seperated`.

Reply:
Good catch, leftover from when I was checking that count matched sums within each function rather than at the end.